app/src/main/java/fr/free/nrw/commons/MediaDataExtractor.java

```diff
@@ -26,23 +34,83 @@ public MediaDataExtractor(MediaClient mediaClient) {
 
     /**
      * Simplified method to extract all details required to show media details.
-     * It fetches media object, deletion status and talk page for the filename
+     * It fetches media object, deletion status, talk page and captions for the filename
      * @param filename for which the details are to be fetched
      * @return full Media object with all details including deletion status and talk page
      */
-    public Single<Media> fetchMediaDetails(String filename) {
+    public Single<Media> fetchMediaDetails(String filename, String pageId) {
         Single<Media> mediaSingle = getMediaFromFileName(filename);
         Single<Boolean> pageExistsSingle = mediaClient.checkPageExistsUsingTitle("Commons:Deletion_requests/" + filename);
         Single<String> discussionSingle = getDiscussion(filename);
-        return Single.zip(mediaSingle, pageExistsSingle, discussionSingle, (media, deletionStatus, discussion) -> {
+        Single<String> captionSingle = getCaption("M"+pageId);
+        Single<JsonObject> depictionSingle = getDepictions(filename);
+        return Single.zip(mediaSingle, pageExistsSingle, discussionSingle, captionSingle, depictionSingle, (media, deletionStatus, discussion, caption, depiction) -> {
             media.setDiscussion(discussion);
+            media.setCaption(caption);
+            media.setDepiction(formatDepictions(depiction));
             if (deletionStatus) {
                 media.setRequestedDeletion();
             }
             return media;
         });
     }
 
+    /**
+     * Obtains captions using filename
+     * @param wikibaseIdentifier
+     *
+     * @return caption for the image in user's locale
+     * Ex: "a nice painting" (english locale) and "No Caption" in case the caption is not available for the image
+     */
+    private Single<String> getCaption(String wikibaseIdentifier) {
+        return mediaClient.getCaptionByWikibaseIdentifier(wikibaseIdentifier);
+    }
+
+    /**
+     * From the Json Object extract depictions into an array list
+     * @param mediaResponse
+     * @return List containing map for depictions, the map has two keys,
+     *  first key is for the label and second is for the url of the item
+     */
+    private ArrayList<Map<String, String>> formatDepictions(JsonObject mediaResponse) {
+        try {
+            JsonArray depictionArray = (JsonArray) mediaResponse.get("Depiction");
+            ArrayList<Map<String, String>> depictedItemList = new ArrayList<>();
+            try {
+                for (int i = 0; i <depictionArray.size() ; i++) {
+                    JsonObject depictedItem = (JsonObject) depictionArray.get(i);
+                    Map <String, String> depictedObject = new HashMap<>();
+                    String label = depictedItem.get("label").toString();
+                    String id =  depictedItem.get("id").toString();
+                    String transformedLabel = label.substring(3, label.length()-3);
```

Comment:
Avoid magic numbers. Define these as constants.

Reply:
Do we need two nested try catch?